keras/engine/data_adapter.py

```diff
@@ -1271,6 +1271,12 @@ def __init__(
         self._insufficient_data = False
         self._model = model
 
+        if steps_per_epoch == 0:
+            raise ValueError(
+                "Got argument `steps_per_epoch=0` passed to `fit()`."
```

Comment:
Let's update this error message to be

"Unexpected value for `steps_per_epoch`. Received value is 0. Please check the docstring for `model.fit()` for supported values."